main.go

```diff
@@ -23,27 +23,31 @@ func main() {
 	if len(os.Args) <= 1 {
 		log.Fatalf("No command passed")
 	}
-
 	environ := os.Environ()
 	if ns, ok := os.LookupEnv(namespaceEnvVar); ok {
 		session := session.Must(session.NewSession())
 		svc := ssm.New(session)
 
-		nsList := strings.Split(ns, ",")
-		for i := range nsList {
-			secrets, err := findSecrets(svc, nsList[i])
-			if err != nil {
-				log.Fatal(err)
-			}
-			environ = addSecrets(environ, secrets)
-		}
+		environ = findAllSecrets(svc, ns, environ)
 	}
 
 	if err := run(os.Args[1:], environ); err != nil {
 		log.Fatal(err)
 	}
 }
 
+func findAllSecrets(svc ssmiface.SSMAPI, nsAll string, environ []string) ([]string) {
+	for _, nsItem := range strings.Split(nsAll, ",") {
+		secrets, err := findSecrets(svc, nsItem)
+		if err != nil {
+			log.Fatal(err)
+		}
+		environ = addSecrets(environ, secrets)
```

Comment:
A small change, but I think it would be nice to build up a map with all the secrets as we iterate over the namespaces, overwriting values as we go. We can then call `addSecrets` once we've collected everything.

Go encourages using the built-in `make` function for initializing maps. You can take a look at [Go maps in action](https://blog.golang.org/go-maps-in-action) to get an idea for the syntax, as well as some background on why Go prefers this option.